x/amm/keeper/route_exact_amount_in.go

```diff
@@ -92,7 +92,7 @@ func (k Keeper) RouteExactAmountIn(
 		// Calculate the total discounted swap fee
 		totalDiscountedSwapFee = totalDiscountedSwapFee.Add(swapFee)
 
-		tokenOutAmount, err = k.SwapExactAmountIn(ctx, sender, actualRecipient, pool, tokenIn, route.TokenOutDenom, _outMinAmount, swapFee)
+		tokenOutAmount, err = k.InternalSwapExactAmountIn(ctx, sender, actualRecipient, pool, tokenIn, route.TokenOutDenom, _outMinAmount, swapFee)
```

Comment:
are we going to integrate discount here or in other function ?

Reply:
@amityadav0 discount will be passed down to the SwapByDenom keeper function using the values returned by the Tier module, this will come as part of a separate PR